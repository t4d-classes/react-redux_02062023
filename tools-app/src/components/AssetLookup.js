import {
  useState,
  useCallback,
  useRef,
  forwardRef,
  useImperativeHandle,
} from 'react';

import { Button, Row, Col, Form } from 'react-bootstrap';



export const AssetLookup = forwardRef((props, ref) => {

  const [
    assetNameInput,
    setAssetNameInput,
  ] = useState('');

  const assetNameInputChange = useCallback((evt) => {
    setAssetNameInput(evt.target.value);
  }, []);

  const assetNameInputElement = useRef(null);

  useImperativeHandle(ref, () => ({
    focus: () => {
      if (assetNameInputElement.current) {
        assetNameInputElement.current.focus();
      }
    },
  }));

  const submitForm = useCallback((evt) => {

    evt.preventDefault();
    setAssetNameInput('');
    props.onSubmit(assetNameInput);

  }, [props, assetNameInput]);

  return (
    <Form style={{ maxWidth: '500px' }} onSubmit={submitForm} aria-label="form">
      <Form.Group as={Row}>
        <Form.Label column>{props.assetTypeLabel}</Form.Label>
        <Col>
          <Form.Control
            type="text"
            ref={assetNameInputElement}
            value={assetNameInput}
            aria-label={`Set ${props.assetTypeLabel}`}
            onChange={assetNameInputChange} />
        </Col>
        <Col>
          <Button variant="primary" type="submit">
            Add to Watch List
          </Button>
        </Col>
      </Form.Group>
    </Form>
  );


});