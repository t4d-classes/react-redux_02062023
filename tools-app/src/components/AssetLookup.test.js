import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { AssetLookup } from './AssetLookup';

test('Snapshot ToolHeader Component', () => {
  expect(
    renderer.create(<AssetLookup buttonText="Submit" />).toJSON(),
  ).toMatchSnapshot();
});

describe('Asset Lookup', () => {
  let submitAssetLookupSpy;

  beforeEach(() => {
    submitAssetLookupSpy = jest.fn();
  });

  const renderComponent = () => {
    render(
      <AssetLookup
        buttonText="Submit"
        onSubmit={submitAssetLookupSpy}
        ref={{ current: null }}
      />,
    );
  };

  test('Submit asset lookup', () => {
    const assetName = 'MSFT';

    renderComponent();

    const assetNameInput = screen.getByRole('textbox');

    fireEvent.change(assetNameInput, { target: { value: assetName } });

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(submitAssetLookupSpy).toHaveBeenCalledWith(assetName);
  });
});
