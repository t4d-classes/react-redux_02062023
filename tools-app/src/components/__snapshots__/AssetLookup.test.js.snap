// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ToolHeader Component 1`] = `
<form
  aria-label="form"
  className=""
  onSubmit={[Function]}
  style={
    Object {
      "maxWidth": "500px",
    }
  }
>
  <div
    className="row"
  >
    <label
      className="form-label col-form-label col"
    />
    <div
      className="col"
    >
      <input
        aria-label="Set undefined"
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="col"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Add to Watch List
      </button>
    </div>
  </div>
</form>
`;
